### PR TITLE
FLI-901: add list and get flags to Java SDK

### PR DESCRIPTION
**Pull request by devumesh**

### CODE REVIEW

File: flipt-java/src/main/java/io/flipt/api/FliptClient.java [hunk 1]
```diff
@@ -22,27 +24,39 @@ private FliptClient(FliptClientBuilder builder) {
...
+            .headers(builder.headers)
+            .setUnhandledExceptionProcessor(builder.unhandledExceptionProcessor)
+            .build();
   }
 
   public Evaluation evaluation() {
     return evaluation;
   }
 
+  public Flag flag() {
```
Comment (markphelps) [inline flipt-java/src/main/java/io/flipt/api/FliptClient.java:41]: would it be weird to rename this method `flags` like we have in the python SDK? https://github.com/flipt-io/flipt-server-sdks/tree/main/flipt-python/flipt/flags

File: flipt-java/src/main/java/io/flipt/api/flags/Flag.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+   * @param namespaceKey the {@link String}
+   * @param flagKey the {@link String}
+   * @param params the {@link CommonParameters}
+   * @return a {@link io.flipt.api.flags.models.Flag} containing the flag information
+   * @throws FlagException if error occurs during the process, such as a network issue or invalid
+   *     request
+   */
+  public io.flipt.api.flags.models.Flag getFlag(
+      String namespaceKey, String flagKey, CommonParameters params) throws FlagException {
+    return this.makeGetRequest(
```
Comment (markphelps) [inline flipt-java/src/main/java/io/flipt/api/flags/Flag.java:114]: we should probably set the namespace to `default` if empty here and below. I'll create an issue to do the same in the Python SDK

**Review by markphelps — COMMENTED** [review]
looks great! ty @devumesh ! just two minor suggestions

File: flipt-java/src/main/java/io/flipt/api/flags/Flag.java (hunk 2 above)
Comment (markphelps) [reply flipt-java/src/main/java/io/flipt/api/flags/Flag.java:114]: #357

File: flipt-java/src/main/java/io/flipt/api/FliptClient.java (hunk 1 above)
Comment (devumesh) [reply flipt-java/src/main/java/io/flipt/api/FliptClient.java]: Hi @markphelps, fixed your the suggestions

**Review by markphelps — APPROVED** [review]
looks great! thank you @devumesh !!

**Pull request merged**